docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,394 @@
 Autograd mechanics
 ==================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note will present an overview of how autograd works. It's not strictly necessary
+to understand all this, but we recommend getting familiar with it, as it will help you
+write more efficient, cleaner programs, and can aid you in debugging.
 
-.. _excluding-subgraphs:
+TL-DR:
 
-Excluding subgraphs from backward
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- Mathematical formulation: we compute vector Jacobian products using the chain rule. We support inplace operations using hidden versioning of the Tensors.
+- Non-gradients: we provide helper functions to be able to compute things that are not real gradients.
+- Will inplace operation work: Because they modify the original value, if anything needs this value for the backward, the backward will fail.
+- Gradients of 0 vs independent: We don't see the difference.
+- Gradients of 0: They can be represented by a Tensor full of ``0`` s, ``None`` in python, undefined tensor in c++ or and error for ``autograd.grad(allow_unused=False)``.
+- Non-differentiable functions: We have many of those. No guarantee but we try to have a good behavior. Fixed rules on what each elementary operation's backward should compute.
```

Comment:
How about this?

"We do not provide anyone  guarantee, but we have fixed rules on what each elementary operation's backward pass should compute which should cover most cases reasonably."